test/Base_Internal_Tests/src/Types/Any_Intersection_Spec.enso

```diff
@@ -0,0 +1,142 @@
+from Standard.Base import all
+from Standard.Test import all
+import Standard.Base.Internal.Meta_Helpers
+
+from project.Types.Any_Intersection import A, B, abc, ac_hidden_b
+
+add_specs suite_builder =
+    suite_builder.group "Any Intersection Tests" group_builder->
+        type_name = Meta_Helpers.simple_type_name_builtin
+
```

Comment:
Defines various groups of four functions and checks how they differ when converting types.